### ISSUE TITLE
[Feature suggestion] Add type and type check to default metadata?

### ISSUE THREAD

**Issue by briochemc** [rafaqz/FieldMetadata.jl#11]
Although I knew about it, the change in the order of the nested metadata declaration just bit me. I almost spent a lot of time debugging some other code when I remembered that updating FieldMetadata.jl was one of the latest things I changed... But if I had done that a few months from now, I would probably have spent a lot of time looking for the mistake.

One possible safeguard could be to have the type declared in some of the default metadata, e.g., `flattenable` could be forced to be a `Bool`, and that type could be checked when creating a new `@flattenable struct`. I.e., 

```julia
@flattenable struct Foo
    a | 3.2
end
```
would throw a `Float64 used in a Bool context` type of error. Just a suggestion, still love this package :)

**Comment by rafaqz** [comment]
Yes sorry about that I'm not really sure about the best way to alert everyone... DiffEq had a warning on package load once, maybe I could have done that but I thought it would just be annoying after you make the change. Hopefully it made the package a bit more intuitive going forward.

I think thats a good idea. We could have a syntax:

```julia
@metadata flattenable Bool
@metadata description String
```

And leaving the type off would be `Any`, just like the current behaviour

Also FYI I figured out we can load and update data metadata from CSV, I think you were interested in that at some stage:
https://github.com/rafaqz/FieldMetadata.jl/issues/8#issuecomment-633353746

**Comment by briochemc** [comment]
Yeah I just thought I'd make the suggestion — glad you like it!

About the CSV load: yes, I am subscribed to this package and I have followed with attention! 😄 It looks great to me! Can't wait for a convenient API to do that 😉 I haven't tried it myself yet because I haven't had the need to reuse a `struct` saved from a previous run since last time. (And last time I ended up simply factoring out the part creating the `struct` out and rerunning that whenever needed, which sort of worked.)
